### PR TITLE
test_data_access init

### PR DESCRIPTION
**Pull request by dannajhun1**

### CODE REVIEW

**Comment by dannajhun1** [comment]
implement unit test for DataAccess class

File: tests/unit_tests/persistence/test__data_access.py [hunk 1]
```diff
@@ -5,3 +5,94 @@
...
+
+        Checks if the data is successfully saved to a JSON file and loaded correctly.
+
+        Args:
+            self: The test case instance.
+        """
+        runnable = "test_runnable"
+        data = {"key": "value"}
+        self.data_access.save_config(runnable, data)
+        expected_file_path = self.data_path / "SZOFTECH" / f"{runnable.replace('/', '_').replace('\'', '_')}.json"
```
Comment (CsullogBeni) [inline tests/unit_tests/persistence/test__data_access.py:31]: There is an error in this line

File: tests/unit_tests/persistence/test__data_access.py [hunk 2]
```diff
@@ -5,3 +5,94 @@
...
+            self: The test case instance.
+        """
+        data = {"key": "value"}
+        self.data_access.save_main_runnables(data)
+        self.data_access.clear_history()
+        expected_file_path = self.data_path / "SZOFTECH" / "main_runnables.json"
+        self.assertFalse(os.path.exists(expected_file_path))
+
+if __name__ == '__main__':
+    unittest.main()
```
Comment (CsullogBeni) [inline tests/unit_tests/persistence/test__data_access.py:98]: There are some warnings in the file, please solve them

**Review by CsullogBeni — CHANGES_REQUESTED** [review]
Have you executed the tests?

**Comment by dannajhun1** [comment]
Fixed asked issues

**Review by CsullogBeni — APPROVED** [review]
Great job! Thank you!

**Pull request merged**